src/EnergyPlus/EvaporativeCoolers.cc

```diff
@@ -240,11 +250,11 @@ namespace EvaporativeCoolers {
             auto const SELECT_CASE_var(EvapCond(EvapCoolNum).EvapCoolerType);
 
             if (SELECT_CASE_var == iEvapCoolerDirectCELDEKPAD) {
-                CalcDirectEvapCooler(EvapCoolNum);
+                CalcDirectEvapCooler(EvapCoolNum, ZoneEvapCoolerPLR);
             } else if (SELECT_CASE_var == iEvapCoolerInDirectCELDEKPAD) {
-                CalcDryIndirectEvapCooler(EvapCoolNum);
+                CalcDryIndirectEvapCooler(EvapCoolNum, ZoneEvapCoolerPLR);
             } else if (SELECT_CASE_var == iEvapCoolerInDirectWETCOIL) {
-                CalcWetIndirectEvapCooler(EvapCoolNum);
+                CalcWetIndirectEvapCooler(EvapCoolNum, ZoneEvapCoolerPLR);
             } else if (SELECT_CASE_var == iEvapCoolerInDirectRDDSpecial) {
```

Comment:
@Nigusse Hi, Bereket, is there any specific reason not to allow researchspecial types cycling with fan as well?  In [these lines](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/EvaporativeCoolers.cc#L3367-L3373), it confused me that it reported power consumption by `FlowRatio` if there's modifier curve (regardless of `PartLoad` which is calculated to meet outlet setpoint), but by `PartLoad` if there's not(regardless of flow ratio). Actually I am not sure what should it be though, the current approach looks confusing to me.

Reply:
The research special evaporative coolers models are different from the other models. They have pump power modifier curve that accounts for pump power calculation as a function of flow fraction. The PartLoad is used only if user did not provide pump power correction curve.

Reply:
Yes that's correct. So if there's no modifier curve, the pump power would only be calculated based on `PartLoad` ignoring mass flow ratio in air loop, so whatever amount of supply air the thermal zones need, it just cycling to meet outlet temperature (supply air temperature). Where it is different with other types is that others always assume a full load outlet temperature. At least in this case ( the `else` condition), it seems to be close to this PR's purpose to fix it. 

In my personal opinion (might be wrong), it seems only consider one of them in each situation looks not enough to include both outlet temperature control (PLR) and mass flow rate control (PLR).

Reply:
I knew that type is different so you didn't touch that in this PR. I mean maybe we could rethink about if we need to change something there as well.